x-pack/plugins/fleet/server/services/policy_watch.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Subscription } from 'rxjs';
+import type {
+  ElasticsearchClient,
+  ElasticsearchServiceStart,
+  KibanaRequest,
+  Logger,
+  SavedObjectsClientContract,
+  SavedObjectsServiceStart,
+} from '@kbn/core/server';
+import { SECURITY_EXTENSION_ID } from '@kbn/core-saved-objects-server';
+import type { ILicense } from '@kbn/licensing-plugin/common/types';
+
+import { pick } from 'lodash';
+
+import type { LicenseService } from '../../common/services/license';
+
+import type { AgentPolicy } from '../../common';
+import { AGENT_POLICY_SAVED_OBJECT_TYPE } from '../../common';
+import {
+  isAgentPolicyValidForLicense,
+  unsetAgentPolicyAccordingToLicenseLevel,
+} from '../../common/services/agent_policy_config';
+
+import { agentPolicyService } from './agent_policy';
+
+export class PolicyWatcher {
+  private logger: Logger;
+  private esClient: ElasticsearchClient;
+  private subscription: Subscription | undefined;
+  private soStart: SavedObjectsServiceStart;
+  constructor(
+    soStart: SavedObjectsServiceStart,
+    esStart: ElasticsearchServiceStart,
+    logger: Logger
+  ) {
+    this.esClient = esStart.client.asInternalUser;
+    this.logger = logger;
+    this.soStart = soStart;
+  }
+
+  /**
```

Comment:
I would be wary of copying big-danger stuff into other plugins, where other teams are principally responsible. 

This was certainly the way to do this a number of years ago when I set this up, but I don't know if this whole copied-over file is still the best way to do it. A lot may have changed in how saved objects can be accessed, permission levels perhaps?

when I learned about how plugins did this in the first place, I grepped the entire kibana codebase for the `LicenseService`. That gave me the list of plugins that had license watchers and did stuff. Then I looked around at their license watcher files, and copied over the bits we need.

Might be worth doing that again to see if any patterns changed. And see if theres a safer way to do saved objects now